apps/openmw/engine.cpp

```diff
@@ -96,7 +96,12 @@ bool OMW::Engine::frame(float frametime)
         // If we are not currently rendering, then RenderItems will not be reused resulting in a memory leak upon changing widget textures (fixed in MyGUI 3.3.2),
         // and destroyed widgets will not be deleted (not fixed yet, https://github.com/MyGUI/mygui/issues/21)
         if (!mEnvironment.getInputManager()->isWindowVisible())
+        {
+            mEnvironment.getSoundManager()->pausePlayback();
             return false;
+        }
+        else
+            mEnvironment.getSoundManager()->resumePlayback();
```

Comment:
Making this call every frame the window is visible seems wasteful. The following call to `SoundManager::update` can probably handle it without needing a redundant call through the interface.

Reply:
In this case we will need to call the `getInputManager()->isWindowVisible()` every frame in the
`SoundManager::update`, so performance will be pretty the same.

Reply:
From what I can see here, `SoundManager::update` doesn't get called if `isWindowVisible` returns false. Is there some other place `SoundManager::update` gets called where that's not true? Alternatively, you could save the result of the `isWindowVisible` call and pass it as a parameter to `SoundManager::update`.

Reply:
> From what I can see here, SoundManager::update doesn't get called if isWindowVisible returns false.

But the SoundManager itself does not know about it, and people who read its code too.